benches/msm.rs

```diff
@@ -0,0 +1,122 @@
+//! This benchmarks Multi Scalar Multiplication (MSM).
+//! It measures `G1` from the BN256 curve.
+//!
+//! To run this benchmark:
+//!
+//!     cargo bench -- msm
+//!
+//! Caveat:  `multicore` should be read as _allowing_ for multicore computation --
+//! not enforcing it.
```

Comment:
If we now have separate benchmarks for single core and multi cores, perhaps just require this benchmark to be ran with `multicore` enabled in `Cargo.toml`?

Reply:
I have refactored the module, and I have updated the caveat accordingly.

Please let me know how you like it now.